### ISSUE TITLE
[8pt] Inundation is routed far from the White Water River but still part of the River

### ISSUE THREAD

**Issue by CarsonPruitt-NOAA** [NOAA-OWP/inundation-mapping#981]

**Comment by CarsonPruitt-NOAA** [comment]
![Image](https://github.com/NOAA-OWP/inundation-mapping/assets/90792257/079feaab-2853-4603-b560-99be6223ca81)

**Comment by CarsonPruitt-NOAA** [comment]
This issue may or may not be caused by the same bug. Please look into it as well. https://vlab.noaa.gov/redmine/issues/119870

**Comment by mluck** [comment]
Discharge in that large catchment is being crosswalked to the mainstem segment highlighted in yellow (NWM streams are blue, DEM-derived reaches in dotted red). Notice that the NWM streams end abruptly (defining the catchment boundary).
<img width="357" alt="image" src="https://github.com/NOAA-OWP/inundation-mapping/assets/6616694/b1f14a01-93d3-477c-af9d-ec99d10db09b">

There doesn't seem to be any visible reason why the NWM streams should end where they do, so this looks like a missing data issue in the NWM streams layer.
<img width="521" alt="image" src="https://github.com/NOAA-OWP/inundation-mapping/assets/6616694/a1b34cec-637c-4efc-b808-81e2d1667a47">

**Comment by CarsonPruitt-NOAA** [comment]
Shouldn't this case be solved by #881? Or does the NWM stream indicate that there should be segments downstream? I.e., the `to` attribute doesn't equal 0.

**Comment by mluck** [comment]
@CarsonPruitt-NOAA You're right, the attribute `to` for those terminal NWM stream segments is 0, so #881 clips the DEM-derived stream from continuing downstream from those segments and the black reaches are removed. However, the red DEM-derived reach downstream is crosswalked to the yellow mainstem NWM stream segment where `to` is not 0 so that's why the DEM-derived stream isn't pruned there.
<img width="301" alt="image" src="https://github.com/NOAA-OWP/inundation-mapping/assets/6616694/f26d073a-b902-4379-9c22-bad3ef1284a7">

**Comment by mluck** [comment]
In the second case, the DEM-derived reach (red) isn't clipped at the end of the NWM stream (blue) because it exceeds the search distance (100 m) from the sink at the end of the NWM stream (distance is 130 m).
<img width="465" alt="image" src="https://github.com/NOAA-OWP/inundation-mapping/assets/6616694/dbfcd8ae-00fc-4585-a559-03a2cc74dd73">
<img width="522" alt="image" src="https://github.com/NOAA-OWP/inundation-mapping/assets/6616694/4056f57e-1887-43d8-8fbf-4e61f0b34ebd">
